db/dcrpg/queries.go

```diff
@@ -3111,6 +3111,41 @@ func appendCoinSupply(charts *cache.ChartData, rows *sql.Rows) error {
 	return nil
 }
 
+// retrieveMissedVotes fetches the missed votes data from the misses and tarnsactions tables.
+func retrieveMissedVotes(ctx context.Context, db *sql.DB, charts *cache.ChartData) (*sql.Rows, error) {
+	rows, err := db.QueryContext(ctx, internal.SelectMissesVotesChartData, charts.MissedVotesTip())
+	if err != nil {
+		return nil, err
+	}
+	return rows, nil
+}
+
+// Append the results from retrieveMissedVotes to the provided ChartData.
+// This is the Appender half of a pair that make up a cache.ChartUpdater.
+func appendMissedVotes(charts *cache.ChartData, rows *sql.Rows) error {
+	defer closeRows(rows)
+	windows := charts.Windows
+	windowSize := uint64(charts.DiffInterval)
+	startHeight := windowSize*uint64(len(windows.MissedVotes)) + uint64(charts.StartPOS)
```

Comment:
Instead of messing with lengths and validations in `charts.go`, it would be easier just to prepend some number of zeroes right here if the `MissedVotes` is empty. 

```
if len(windows.MissedVotes)  == 0 {
    windows.MissedVotes = append(windows.MissedVotes, make([]int64, charts.StartPOS / charts.DiffInterval)...)
}
```
*untested

It's even technically correct that there were zero missed tickets during those windows.